src/IW/Sync/Search.hs

```diff
@@ -1,97 +1,214 @@
+    SearchResult count vec <- executeGithubSearch paths properties firstDay recent 1
+    let firstPage = V.toList vec
+        -- | If count is 0, then all results for the given properties have been obtained.
+    if | count == 0   -> do
+            log I "All results obtained"
+            pure acc
+        -- | If count is less than or equal to 1000, then the interval is good and we can get
+        -- the rest of the results on pages 2 to 10.
+       | count <= 1000 -> do
+            listOfSearchResult <- mapM (executeGithubSearch paths properties firstDay recent) [2..10]
+            let remainingPages = concatMap searchResultToList listOfSearchResult
+            -- | Recursive call with a new @recent@ arguement and a new @acc@ arguement
+            -- representing all pages accumulated up to this point.
+            githubSearchAll paths properties (pred firstDay) interval (firstPage <> remainingPages <> acc)
+        -- | Otherwise, call the function with the same arguments but a smaller interval.
+       | otherwise     -> githubSearchAll paths properties recent (pred interval) acc
+  where
+    -- | The first day of the search interval. It's calculated by subtracting the size of the
+    -- interval from the most recent day of the search interval.
+    firstDay :: Day
+    firstDay = negate interval `addDays` recent
 
--- | Search all repositories built with the Haskell language.
-searchHaskellRepos :: IO (Either GitHub.Error (SearchResult GitHub.Repo))
-searchHaskellRepos = searchRepos "language:haskell"
+    searchResultToList :: SearchResult a -> [a]
+    searchResultToList (SearchResult _ vec) = V.toList vec
 
--- | Construct a github search query from a list of labels.
-labelsToSearchQuery :: [Label] -> Text
-labelsToSearchQuery = foldMap (\Label{..} -> "label:\"" <> unLabel <> "\" ")
+-- | Executes a GitHub search action within the context of the @App@ monad.
+executeGithubSearch
+    :: forall a env m.
+       ( MonadIO m
+       , MonadUnliftIO m
+       , WithError m
+       , WithLog env m
+       , FromJSON a
+       )
+    => Paths -- ^ Query paths
+    -> Text  -- ^ Query properties
+    -> Day   -- ^ The first day of the date interval
+    -> Day   -- ^ The last day of the date interval
+    -> Int   -- ^ The number of the page to be searched
+    -> m (SearchResult a)
+executeGithubSearch paths properties from to page = do
+    searchLimit <- getSearchRateLimit
+    log I $ "Current search limit information: " <> show searchLimit
+    if limitsRemaining searchLimit > 0
+        then do
+            log D $ "Searching GitHub API with following request: " <> showGithubQuery paths properties from to page
+            liftIO (githubSearch paths properties from to page) >>= \case
+                Left err -> throwError $ githubErrToAppErr err
+                Right searchRes -> pure searchRes
+        else do
+            log I "API limit reached. Delaying search..."
+            threadDelay 60000000
```

Comment:
Let's just throw an error here. The limits for GitHub API is 5000 per hour. So waiting one minute doesn't really help... If we hit the limit, we will handle it separately. You can create the issue to handle cases when we reach rate limit.